Automattic/Gravatar-SDK-iOS
# Add Placeholder UI & loading state & profile view controller

Closes https://github.com/Automattic/Gravatar-SDK-iOS/issues/213 https://github.com/Automattic/Gravatar-SDK-iOS/issues/211 https://github.com/Automattic/Gravatar-SDK-iOS/issues/210 https://github.com/Automattic/Gravatar-SDK-iOS/issues/209 https://github.com/Automattic/Gravatar-SDK-iOS/issues/208 

### Description

- Adds ghost ui. I designed the placeholder and the activity indicator as separate things. Placeholder can be viewed without the loading animations(for example if the service call fails, the placeholder will remain but the animations will stop). One can also customize the placeholder & activity indicator.
- I didn't set the bottom sheet to use intrinsic size in this PR. It uses predefined size.

<img width=250 src="https://github.com/Automattic/Gravatar-SDK-iOS/assets/5032900/452a36a0-439e-40e7-a4c4-26848999edd0" />


<img width=250 src="https://github.com/Automattic/Gravatar-SDK-iOS/assets/5032900/51ff185f-08f4-49bc-b3b2-f2d5148c6938" />



- Adds a ProfileViewController to test the standalone usage of our profile views. ProfileViewController does the service call and emits the view model state changes & updates the view.

### Testing Steps

Check these 2 pages in the demo app:

- Profile View Controller Presentations
- Profile Card
